# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/target/lvp/patches/xorg_hotfixes.patch
# ROCK Linux is Copyright (C) 1998 - 2005 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- xc/extras/ogl-sample/main/gfx/lib/glu/libnurbs/internals/mymath.h	2005-01-16 17:40:50.170615424 +0100
+++ xc/extras/ogl-sample/main/gfx/lib/glu/libnurbs/internals/mymath.h	2005-01-16 17:40:57.856447000 +0100
@@ -66,8 +66,8 @@ extern "C" float	floorf(float);
 #define NEEDCEILF
 #endif
 #elif defined(NEEDCEILF)
-extern "C" float	ceilf(float);
-extern "C" float	floorf(float);
+//extern "C" float	ceilf(float);
+//extern "C" float	floorf(float);
 #endif
 
 #ifdef LIBRARYBUILD
--- xc/programs/glxinfo/Imakefile	2004-04-23 21:54:33.000000000 +0200
+++ xc/programs/glxinfo/Imakefile	2005-01-17 10:53:13.084953608 +0100
@@ -5,7 +5,7 @@ XCOMM $XFree86: xc/programs/glxinfo/Imak
 #if BuildGLULibrary
 
         DEPLIBS = $(DEPGLULIB) $(DEPGLXLIB) $(DEPXLIB)
-LOCAL_LIBRARIES = $(GLULIB) $(GLXLIB) $(XLIB) 
+LOCAL_LIBRARIES = $(GLULIB) $(GLXLIB) $(XLIB) -ldl -lXxf86vm
         DEFINES = -DDO_GLU
 
 #else
--- xc/programs/glxgears/Imakefile	2004-09-02 07:50:38.000000000 +0200
+++ xc/programs/glxgears/Imakefile	2005-01-17 11:39:49.878776320 +0100
@@ -8,7 +8,7 @@ XCOMM $XFree86: xc/programs/glxinfo/Imak
 #endif
 
         DEPLIBS = $(DEPGLXLIB) $(DEPXPLIB) $(DEPXLIB)
-LOCAL_LIBRARIES = $(GLXLIB) $(XPRINTLIBS) $(XLIB)
+LOCAL_LIBRARIES = $(GLXLIB) $(XPRINTLIBS) $(XLIB) -ldl -lXxf86vm
         DEFINES = $(XPRINTDEFINES)
   SYS_LIBRARIES = MathLibrary
 
--- xc/programs/xdriinfo/Imakefile	2005-01-18 10:16:29.014022152 +0100
+++ xc/programs/xdriinfo/Imakefile	2005-01-18 10:16:37.274766328 +0100
@@ -4,7 +4,7 @@ XCOMM $XFree86$
 
             SRCS = xdriinfo.c
             OBJS = xdriinfo.o
- LOCAL_LIBRARIES = $(GLXLIB) $(XLIB) 
+ LOCAL_LIBRARIES = $(GLXLIB) $(XLIB)  -ldl -lXxf86vm
          DEPLIBS =
 
 AllTarget(ProgramTargetName(xdriinfo))
--- xc/fonts/util/ucs2any.c	2004-08-26 13:58:06.000000000 +0200
+++ xc/fonts/util/ucs2any.c	2005-01-18 12:19:47.123338584 +0100
@@ -61,7 +61,7 @@
 /* global variable for argv[0] */
 const char *my_name = NULL;
 
-#ifdef NEED_BASENAME
+#if 0
 static char *
 basename(char *pathname)
 {
--- xc/programs/fc-list/Imakefile	2005-01-18 14:02:56.000000000 +0100
+++ xc/programs/fc-list/Imakefile	2005-01-18 15:15:48.274799520 +0100
@@ -4,8 +4,8 @@ FONTCONFIGDIR = $(TOP)/extras/fontconfig
 
 INCLUDES=$(FREETYPE2INCLUDES) -I$(FONTCONFIGDIR)
 
-LOCAL_LIBRARIES=FontconfigClientLibs
-        DEPLIBS=FontconfigClientDepLibs
+LOCAL_LIBRARIES=FontconfigClientLibs -lz
+        DEPLIBS=FontconfigClientDepLibs -lz
 
 SRCS=fc-list.c
 OBJS=fc-list.o
--- xc/programs/x11perf/Imakefile	2005-01-18 13:22:58.000000000 +0100
+++ xc/programs/x11perf/Imakefile	2005-01-18 16:13:52.097178016 +0100
@@ -40,7 +40,7 @@ XCOMM $XFree86: xc/programs/x11perf/Imak
 		  do_dots.o do_windows.o do_movewin.o do_text.o \
 		  do_blt.o do_arcs.o \
 		  do_tris.o do_complex.o do_traps.o
-LOCAL_LIBRARIES = $(XFTLIBS) $(XRENDERLIBS) $(XMUULIB) $(XLIB)
+LOCAL_LIBRARIES = $(XFTLIBS) $(XRENDERLIBS) $(XMUULIB) $(XLIB) -lz
         DEPLIBS = $(XFTDEPS) $(XRENDERDEPS) $(DEPXMUULIB) $(DEPXLIB)
   SYS_LIBRARIES = MathLibrary
 
--- xc/programs/xclock/Imakefile	2005-01-19 10:15:42.031830344 +0100
+++ xc/programs/xclock/Imakefile	2005-01-19 10:16:01.127927296 +0100
@@ -15,7 +15,7 @@ XCOMM $XFree86: xc/programs/xclock/Imake
     XRENDERINCS = $(XFTINCLUDES) $(XRENDERINCLUDES) -I$(XBUILDINCDIR)
 #endif
         DEPLIBS = $(XRENDERDEPS) XkbClientDepLibs XawClientDepLibs
-LOCAL_LIBRARIES = $(XRENDERLIBS) XkbClientLibs XawClientLibs
+LOCAL_LIBRARIES = $(XRENDERLIBS) XkbClientLibs XawClientLibs -lz
            SRCS = xclock.c Clock.c
            OBJS = xclock.o Clock.o
         HEADERS = ClockP.h Clock.h
--- xc/programs/xfd/Imakefile	2004-04-23 21:54:47.000000000 +0200
+++ xc/programs/xfd/Imakefile	2005-01-19 12:19:14.690934872 +0100
@@ -15,7 +15,7 @@ XCOMM $XFree86: xc/programs/xfd/Imakefil
     XRENDERINCS = $(XFTINCLUDES) $(XRENDERINCLUDES) -I$(XBUILDINCDIR)
 #endif
         DEPLIBS = $(XRENDERDEPS) XawClientDepLibs XkbClientDepLibs
-LOCAL_LIBRARIES = $(XRENDERLIBS) XawClientLibs XkbClientLibs
+LOCAL_LIBRARIES = $(XRENDERLIBS) XawClientLibs XkbClientLibs -lz
            SRCS = xfd.c grid.c
            OBJS = xfd.o grid.o
        INCLUDES = $(XRENDERINCS)
